src/main/java/Library.java

```diff
@@ -0,0 +1,123 @@
+                name = scanner.next();
+                deleteBookFromCatalog();
+                bool = true;
+            }
+            if (tittle == 3) {
+                scanner.close();
+                bool = false;
+            }
+        }while (bool);
+    }
+
+    public static void deleteBookFromCatalog(){
+
+        Books book = new Books(name);
+        HashSet<Object> library = null;
+        try {
+            library = reedingCatalog();
+        } catch (IOException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        assert library != null;
+        library.remove(book);
+        try {
+            File file = new File("src/main/resources/library/books catalog");
+            FileOutputStream fileOut = new FileOutputStream(file);
+            ObjectOutputStream out = new ObjectOutputStream(fileOut);
+            for (Object s : library) out.writeObject(s);
+            out.close();
+            fileOut.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void saveBookToCatalog() {
+
+        Books book = new Books(name, author, categorize);
+
+        HashSet<Object> library = null;
+        try {
+            library = reedingCatalog();
+        } catch (IOException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        assert library != null;
+        library.add(book);
+        try {
+            File file = new File("src/main/resources/library/books catalog");
+            FileOutputStream fileOut = new FileOutputStream(file);
+            ObjectOutputStream out = new ObjectOutputStream(fileOut);
+            for (Object s : library) out.writeObject(s);out.close();
+
+            fileOut.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static HashSet<Object> reedingCatalog() throws IOException, ClassNotFoundException {
+        try {
+            File file = new File("src/main/resources/library/books catalog");
+            FileInputStream fileInput = new FileInputStream(file);
+            ObjectInputStream input = new ObjectInputStream(fileInput);
+            FileReader fileReader = new FileReader(file);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            HashSet<Object> newLibrary = new HashSet<>();
+            try {
+                do {
+                    if (file.canRead()){
+                        newLibrary.add(input.readObject());
+                    }else System.out.println("File is empty!");
+                }while (bufferedReader.ready());
+            }catch (EOFException ignored){
+            }
+            bufferedReader.close();
+            fileReader.close();
+            input.close();
+            fileInput.close();
+            return newLibrary;
+        }catch (IOException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
returning null considered generally a bad practice, some time you can do it (if you don't have another option)